Controller/Webhook/Order.php

```diff
@@ -127,27 +127,28 @@ public function execute()
             $gr4vy_transaction_id = $request->target->id;
             $statuses = $this->gr4vyOrder->getGr4vyTransactionStatuses();
 
-            if ($gr4vy_status = $this->transactionApi->getStatus($gr4vy_transaction_id)) {
-                $dataModel = $this->transactionRepository->getByGr4vyTransactionId($gr4vy_transaction_id);
+            $transactionDetail = $this->transactionApi->getTransactionDetail($gr4vy_transaction_id);
+
+            if ($gr4vy_status = $transactionDetail["status"]) {
 
```

Comment:
@steve-gr4vy why did we move the if statement to after the fetch?

Reply:
In the previous code we were calling 'GET /transaction/{id}' endpoint twice - once from the 'getStatus()' helper function and then again from the 'getTransactionDetail()' helper function - this is the same logic but calls the endpoint only once